### ISSUE TITLE
Pioneer does not run with Lua-5.3

### ISSUE THREAD

**Issue by sagitter** [pioneerspacesim/pioneer#3712]
Hi all,

I'm trying to build `Pioneer` against external `Lua` (5.3.2) libraries on `Fedora` 23; unfortunately, it does not run:

```
$ export force_s3tc_enable=true
$ pioneer
ver yes (yes) on: Linux

System Name: Linux
Host Name: localhost.localdomain
Release(Kernel) Version: 4.4.9-300.fc23.x86_64
Kernel Build Timestamp: #1 SMP Wed May 4 23:56:27 UTC 2016
Machine Arch: x86_64
Domain Name: (none)

ATTENTION: default value of option force_s3tc_enable overridden by environment.
SDL Version 2.0.4
Initialized OpenGL 3.1, with extensions, renderer
invalid axis binding '' in config file for BindAxisPitch
invalid axis binding '' in config file for BindAxisRoll
invalid axis binding '' in config file for BindAxisYaw
started 1 worker threads
ShipType::Init()
Lua::Init()
Server agent disabled
GalaxyGenerator::Init()
Creating new galaxy generator 'legacy' version 1
error: [string "[T] @systems/00_sol.lua"]:141: bad argument #1 to 'f' (number has no integer representation)
stack traceback:
	[C]: in ?
	[C]: in function 'f'
	[string "[T] @systems/00_sol.lua"]:141: in main chunk

```

Pioneer runs fine if built against distributed Lua files in `contrib/` directory.
Given that I need to unbundle libraries from Pioneer source code as much as possible, can you fix this issue please ?

**Comment by fluffyfreak** [comment]
@sagitter the pioneer-thirdparty exists so that we don't have to unbundle things. 
Think of it this way, we fix this so that it works with Lua 5.x.x on Fedora 23, and it stops working elsewhere, but by your own admission it works with our pioneer-thirdparty.

Why must you unbundle it?

The Lua folks who might best be able to help are probably @laarmen, @impaktor or @robn if they want to chip in.

**Comment by fluffyfreak** [comment]
Actually let me try again, we bundle Lua with Pioneer because that's the version that we use and compile against, not even as part of `pioneer-thirdparty` but in with the codebase itself.

Are you trying to remove it from Pioneer itself so you can use whatever version is installed on the OS/Distribution that you're using?

**Comment by sagitter** [comment]
> @sagitter the pioneer-thirdparty exists so that we don't have to unbundle things.
> Think of it this way, we fix this so that it works with Lua 5.x.x on Fedora 23, and it stops working elsewhere, but by your own admission it works with our pioneer-thirdparty.
> 
> Why must you unbundle it?

Because it is a required condition to include a software into Fedora build-system. I understand your point of view, I had to point out this situation.

> Are you trying to remove it from Pioneer itself so you can use whatever version is installed on the OS/Distribution that you're using?

Exactly; if you say me that unbundling is not expected yet, I can also build `Pioneer` against "your" Lua.

**Comment by fluffyfreak** [comment]
Ah I see. Well we're definitely not expecting people to unbundle the version that we use, I think as you have found that it will not work with 5.3.2

**Comment by laarmen** [comment]
Oh well. It appears Fedora has Lua 5.1 (compat-lua) and 5.3 but not 5.2.

As far as I am concerned, we're not planning on porting to 5.3 anytime soon. As you might have noticed, we bundle Lua directly in our main git repo: we're assuming we'll run on this version and not another. We might be compatible with other versions of the 5.2 branch, but that would be coincidental.

The port to 5.3 is not trivial to do, and if we were to do any porting I'd rather do it to LuaJIT (which, in terms of language stands between 5.1 and 5.2)

**Comment by fluffyfreak** [comment]
I'm closing this now as no-one on the team has expressed interest in either porting us to Lua 5.3 or to anything else and it seems like a bit of a none issue.

**Comment by ecraven** [comment]
I've tried with luajit, which is lua 5.1. We use lua 5.2, so we'd have to backport a few things, it is *not* just a drop-in replacement... I think the question is do we want to go forward to lua 5.3, or "sideways" to luajit?

**Comment by laarmen** [comment]
Do we want to move at all? The moment we have performance issues caused by
Lua, this should be considered, but I'm not sure we should bother before
that.

**Comment by ecraven** [comment]
I agree, however moving all the ui into lua, I'm almost certain we'll want to move at some point (and probably to luajit, not sure whether 5.3 is any faster than 5.2).
